skill/scenes.py

```diff
@@ -116,6 +117,155 @@ def handle_local_intents(self, request: Request):
+    def reply(self, request: Request):
+        text, tts = texts.start_setting()
+        return self.make_response(request, text, tts, buttons=[button("Помощь")])
+
+    def handle_local_intents(self, request: Request):
+        if intents.FIO in request.entities_list:
+            return Settings_GetSchool()
+
+
+class Settings_GetSchool(GlobalScene):
+    def reply(self, request: Request):
+        name = request.entity(intents.FIO)[0]["first_name"].capitalize()
+
+        text, tts = texts.what_school(name)
+        return self.make_response(
+            request,
+            text,
+            tts,
+            buttons=[button("Помощь")],
+            state={state.TEMP_NAME: name},
+        )
+
+    def handle_local_intents(self, request: Request):
+        if intents.NUMBER in request.entities_list:
+            return Settings_GetClassNumber()
+
+
+class Settings_GetClassNumber(GlobalScene):
+    def reply(self, request: Request):
+        school_num = str(request.entity(intents.NUMBER)[0])
+        text, tts = texts.what_classnumber()
+        return self.make_response(
+            request,
+            text,
+            tts,
+            buttons=[button("Помощь")],
+            state={state.TEMP_SCHOOL: str(school_num)},
+        )
+
+    def handle_local_intents(self, request: Request):
+        if intents.NUMBER in request.entities_list:
+            class_num = request.entity(intents.NUMBER)[0]
+            if 1 <= class_num <= 11:
+                class_letter = request.tokens[-1]
+                if "А" <= class_letter.capitalize() <= "Я":
+                    return Settings_Confirm()
+                else:
+                    return Settings_GetClassLetter()
+
+
+class Settings_GetClassLetter(GlobalScene):
+    def reply(self, request: Request):
+        class_num = request.entity(intents.NUMBER)[0]
+        text, tts = texts.what_classletter()
+        return self.make_response(
+            request,
+            text,
+            tts,
+            buttons=[
+                button("А"),
+                button("Б"),
+                button("В"),
+                button("Г"),
+                button("Д"),
+                button("Помощь"),
+            ],
+            state={state.TEMP_CLASS_ID: str(class_num)},
+        )
+
+    def handle_local_intents(self, request: Request):
+        class_letter = request.tokens[-1]
+        if "А" <= class_letter.capitalize() <= "Я":
+            return Settings_Confirm()
+
+
+class Settings_Confirm(GlobalScene):
+    def reply(self, request: Request):
+        prev_session = request.session.get("scene")
+        if prev_session == "Settings_GetClassNumber":  # назвали класс полностью
+            class_num = request.entity(intents.NUMBER)[0]
+            class_letter = request.tokens[-1]
+        else:
+            class_num = request.session.get(state.TEMP_CLASS_ID)
+            class_letter = request.tokens[-1]
+        class_id = str(class_num) + class_letter.capitalize()
```

Comment:
```suggestion
        class_id = str(class_num) + class_letter.upper()
```
Хотел еще предложить сохранять цифру и букву отдельно, в каком-нибудь датаклассе, но потом подумал - так действительно будет проще. Буква ведь всегда 1, так-что распарсить если понадобится не будет проблемой.